src/app/shared/components/header/header.component.ts

```diff
@@ -8,13 +8,21 @@ import { Subscription } from 'rxjs';
   templateUrl: './header.component.html',
   styleUrls: ['./header.component.scss']
 })
-export class HeaderComponent implements OnDestroy {
+export class HeaderComponent implements OnInit, OnDestroy {
   currentUser: User;
   subscription: Subscription;
 
   constructor(private authService: AuthService) {
-    this.subscription = authService.getAuthUser()
-      .subscribe( user => this.currentUser = user );
+    
+    this.subscription = authService.getAuthUserObservable()
+      .subscribe( user => {
+        console.log('constuctor: user=', user);
+        this.currentUser = user;
```

Comment:
If you want to receive "fresh" user after login/logout operations, you should update logic inside your auth service to emit new event after something was happened. (only `of(currentUser)` is not enough, because you emit event only once)